validation/markets-validator.js

```diff
@@ -0,0 +1,597 @@
+            this.errors.push(
+                "Attribute 'coordinates' must contain two values not " + coordinates.length + ".");
+        } else {
+            var lon = coordinates[0];
+            if (!longitudeInValidRange(lon)) {
+                this.errors.push(new LongitudeRangeExceedanceIssue("coordinates[0]", lon));
+            }
+            var lat = coordinates[1];
+            if (!latitudeInValidRange(lat)) {
+                this.errors.push(new LatitudeRangeExceedanceIssue("coordinates[1]", lat));
+            }
+        }
+    };
+
+    this.validateZoomLevel = function(zoomLevel) {
+        if (zoomLevel === undefined) {
+            this.errors.push(new UndefinedAttributeIssue("zoom_level"));
+        } else if (zoomLevel === null) {
+            this.errors.push(new NullAttributeIssue("zoom_level"));
+        } else if (zoomLevel < 1 || zoomLevel > 18) {
+            this.errors.push(new RangeExceedanceIssue("zoom_level", 1, 18, zoomLevel));
+        }
+    };
+
+    this.getErrorsCount = function() {
+        return this.errors.length;
+    };
+
+    this.getWarningsCount = function() {
+        return this.warnings.length;
+    };
+
+    this.hasErrors = function() {
+        return this.errors.length > 0;
+    };
+
+    this.hasWarnings = function() {
+        return this.warnings.length > 0;
+    };
+
+}
+
+
+function latitudeInValidRange(lat) {
+    return (lat > MIN_LATITUDE && lat <= MAX_LATITUDE);
+}
+
+function longitudeInValidRange(lon) {
+    return (lon > MIN_LONGITUDE && lon <= MAX_LONGITUDE);
+}
+
+
+function CustomIssue(message) {
+
+    this.message = message;
+
+    this.toString = function() {
+        return this.message;
+    };
+}
+
+function UndefinedAttributeIssue(attributeName) {
+
+    this.attributeName = attributeName;
+
+    this.toString = function() {
+        return "Attribute '" + this.attributeName + "' cannot be undefined.";
+    };
+}
+
+
+function NullAttributeIssue(attributeName) {
+
+    this.attributeName = attributeName;
+
+    this.toString = function() {
+        return "Attribute '" + this.attributeName + "' cannot be null.";
+    };
+}
+
+function EmptyAttributeIssue(attributeName) {
+
+    this.attributeName = attributeName;
+
+    this.toString = function() {
+        return "Attribute '" + this.attributeName + "' cannot be empty.";
+    };
+}
+
+function EmptyObjectIssue(attributeName) {
+
+    this.attributeName = attributeName;
+
+    this.toString = function() {
+        return "Attribute '" + this.attributeName + "' cannot be an empty object.";
+    };
+}
+
+function LatitudeRangeExceedanceIssue(attributeName, actual) {
+    return new RangeExceedanceIssue(attributeName, MIN_LATITUDE, MAX_LATITUDE, actual);
+}
+
+function LongitudeRangeExceedanceIssue(attributeName, actual) {
+    return new RangeExceedanceIssue(attributeName, MIN_LONGITUDE, MAX_LONGITUDE, actual);
+}
+
+function RangeExceedanceIssue(attributeName, min, max, actual) {
```

Comment:
I disagree at this point. Code should be self documenting in the first place. Class or method names should describe the intend, same for parameters. I tried to satisfy this aim. Redundant comments will make the code less readable.